projects/ngx-d3timeline/src/lib/ngx-d3timeline.component.ts

```diff
@@ -103,10 +102,15 @@ export class NgxD3timelineComponent implements OnInit, AfterViewInit {
 
   @ViewChild('svgEl') svgEl: ElementRef<SVGElement>;
 
-  constructor(public timeline: NgxD3TimelineService) {}
+  constructor(
+    public timeline: NgxD3TimelineService,
+    private hostElement: ElementRef,
+    private changeDetector: ChangeDetectorRef
+  ) {}
 
   ngOnInit(): void {
     this.timeline.onActivityDropped(this.activityDropped);
+    this.timeline.setupResizing(this.hostElement, this.changeDetector);
```

Comment:
What do you think about having an `ngOnitInit` in the service and this line can be in there?

Reply:
moved to service constructor (no lifecycle events in service except OnDestroy)

Reply:
My bad, I thought that since there's an onDestroy there would also be an onInit.  

Feels like too much work is being in this function for it to go in the constructor. I think onInit is the correct place for this.

Reply:
done